pyproject.toml

```diff
@@ -18,27 +18,30 @@ dependencies = [
     "ipython",
     "kafka-python",
     "matplotlib",
-    "mplcursors",
-    "mpl-pan-zoom",
     "numpy",
-    "openpyxl",
     "pandas<2",
-    "PyQt5",
     "PyYAML",
-    "pyflakes",
-    "QScintilla==2.13",
     "xarray",
     "scipy",
     "supervisor",
     "termcolor",
     "tomli",
     "pasha",
     "extra_geom",
-    "zulip",
-    "tabulate"
 ]
 
 [project.optional-dependencies]
+gui = [
+    "adeqt",
+    "mplcursors",
+    "mpl-pan-zoom",
+    "openpyxl",  # for spreadsheet export
+    "PyQt5",
+    "pyflakes",  # for checking context file in editor
+    "QScintilla==2.13",
+    "tabulate",
```

Comment:
Unrelated to the PR, but we can actually remove this and a couple other packages too :sweat_smile: It looks like tabulate was added when we implemented Zulip support but it was never used.

The other packages that can be removed are:
- PyYAML
- tomli
- pasha
- extra_geom
- euxfel_bunch_pattern

They were added to be used in context files before we had support for separate environments.

Reply:
I thought this for tabulate, but it's actually required for pandas' `to_markdown` method. I think you're right about the others, though.